postgres/datadog_checks/postgres/statements.py

```diff
@@ -95,32 +91,47 @@ def _get_pg_stat_statements_columns(self, db):
         version is not a reliable way to determine the available columns on `pg_stat_statements`. The database can
         be upgraded without upgrading extensions, even when the extension is included by default.
         """
+        if len(self._stat_column_cache) == 0:
```

Comment:
We should do early return here and have all other parts of the code call `_get_pg_stat_statements_columns` instead of accessing `self._stat_column_cache` directly, else it's hard to reason about whether the other parts of the code are accessing the cache before or after initialization. 

```python
if self._stat_column_cache:
    return self._stat_column_cache

... initialize
```